src/udev/udev-builtin-hwdb.c

```diff
@@ -77,7 +77,7 @@ static const char *modalias_usb(struct udev_device *dev, char *s, size_t size) {
         pn = strtol(p, NULL, 16);
         if (pn <= 0)
                 return NULL;
-        snprintf(s, size, "usb:v%04Xp%04X*", vn, pn);
+        snprintf(s, size, "usb:v%04Xp%04X*", (uint16_t)vn, (uint16_t)pn);
```

Comment:
Please add a space after the case: "(uint16_t) vn". And a comment above the snprintf. We are trying to get rid of all casts in the codebase, and without a comment I'm afraid somebody will remove it.